### ISSUE TITLE
Weighted SSL model not unfreezable

### ISSUE THREAD

**Issue by pplantinga** [speechbrain/speechbrain#2759]
### Describe the bug

In our HF Weighted SSL model implementation, we `detach()` the hidden states, meaning weights are not updated.

Relevant code:
https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/lobes/models/huggingface_transformers/weighted_ssl.py#L81

```
hidden_states = torch.stack(feats.hidden_states, dim=0).detach()
```

### Expected behaviour

If passing `freeze=False` I'd expect the weights to get updated.

**Comment by pplantinga** [comment]
Also if `layernorm=True` then the hidden states are converted to a list which causes a program crash. They should be re-stacked into a tensor.
